packages/ui-react/src/containers/ShelfList/ShelfList.tsx

```diff
@@ -1,88 +1,136 @@
 import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
+import { useTranslation } from 'react-i18next';
 import { shallow } from '@jwp/ott-common/src/utils/compare';
 import InfiniteScroll from 'react-infinite-scroller';
+import { useQueries, useQueryClient } from 'react-query';
 import type { Content } from '@jwp/ott-common/types/config';
 import { useAccountStore } from '@jwp/ott-common/src/stores/AccountStore';
 import { useConfigStore } from '@jwp/ott-common/src/stores/ConfigStore';
-import { useWatchHistoryStore } from '@jwp/ott-common/src/stores/WatchHistoryStore';
+import { useWatchHistoryStore, useWatchHistoryStore as WatchHistoryStore } from '@jwp/ott-common/src/stores/WatchHistoryStore';
+import { useFavoritesStore as FavoritesStore } from '@jwp/ott-common/src/stores/FavoritesStore';
 import { slugify } from '@jwp/ott-common/src/utils/urlFormatting';
-import { parseAspectRatio, parseTilesDelta } from '@jwp/ott-common/src/utils/collection';
-import { testId } from '@jwp/ott-common/src/utils/common';
-import { PersonalShelf } from '@jwp/ott-common/src/constants';
+import { generatePlaylistPlaceholder, parseAspectRatio, parseTilesDelta } from '@jwp/ott-common/src/utils/collection';
+import { isTruthyCustomParamValue, testId } from '@jwp/ott-common/src/utils/common';
+import { PersonalShelf, PLAYLIST_LIMIT } from '@jwp/ott-common/src/constants';
+import { getModule } from '@jwp/ott-common/src/modules/container';
+import ApiService from '@jwp/ott-common/src/services/ApiService';
+import type { Playlist } from '@jwp/ott-common/types/playlist';
+import { isScheduledOrLiveMedia } from '@jwp/ott-common/src/utils/liveEvent';
 
 import ShelfComponent from '../../components/Shelf/Shelf';
 import InfiniteScrollLoader from '../../components/InfiniteScrollLoader/InfiniteScrollLoader';
-import PlaylistContainer from '../PlaylistContainer/PlaylistContainer';
+import ErrorPage from '../../components/ErrorPage/ErrorPage';
 
 import styles from './ShelfList.module.scss';
 
-const INITIAL_ROW_COUNT = 6;
+const INITIAL_ROWS_LOADED = 6;
 const LOAD_ROWS_COUNT = 4;
 
 type Props = {
   rows: Content[];
 };
 
+const placeholderData = generatePlaylistPlaceholder(30);
+
 const ShelfList = ({ rows }: Props) => {
   const { accessModel } = useConfigStore(({ accessModel }) => ({ accessModel }), shallow);
-  const [rowCount, setRowCount] = useState(INITIAL_ROW_COUNT);
+  const [rowsLoaded, setRowsLoaded] = useState(INITIAL_ROWS_LOADED);
+  const { t } = useTranslation('error');
 
   const watchHistoryDictionary = useWatchHistoryStore((state) => state.getDictionaryWithSeries());
 
   // User
   const { user, subscription } = useAccountStore(({ user, subscription }) => ({ user, subscription }), shallow);
 
+  // Todo: move to more common package?
```

Comment:
If someone has a good and relatively easy suggestion then I'm happy to do this now.

Reply:
We could encapsulate this in a `usePlaylists` or `useShelvesData` hook?

Reply:
Yes, that would make it react (including react-native), perhaps that's enough for now? Easy implementation indeed.